src/dfcx_scrapi/tools/analysis_util.py

```diff
@@ -0,0 +1,162 @@
+                    distance[row][col-1] + 1,
+                    distance[row-1][col-1] + cost
+                )
+
+        # Computation of the Levenshtein Distance Ratio
+        ratio = ((len(s)+len(t)) - distance[row][col]) / (len(s)+len(t))
+        return ratio
+
+
+    @staticmethod
+    def calc_tp_distances(
+        intent_key,
+        intent_comparator,
+        threshold=0.75,
+        silent=False
+    ):
+        '''
+        compares the training phrases between two intents to find the
+        levenshtein distance between all training phrases. Structure of the
+        returned object depends on which intent is the key, and which is the
+        comparator.
+
+        Args:
+          intent_key: intent protobuf or dataframe object. In the returned
+          object, the utterances of this intent are paired with a list of
+          utterances from the comparator.
+
+          intent_comparator: intent protobuf or dataframe object. In the
+          returned object, the utterances of this intent will be included
+          in a list associated with every phrase whose similarity ratio
+          is above the defined threshold.
+
+          threshold: float describing the levenshtein distance above which
+          a pair of phrases should be associated. Default: .75
+
+        Returns:
+          tp_distances: Object listing each training phrase in intent_key.
+          Associated with each intent_key will be a list of pairings,
+          where each pair represents an utterance from intent_comparator
+          with a similarity ratio above the defined threshold.
+        '''
+
+        if isinstance(intent_key, types.Intent):
+            list_keys = Intents.intent_proto_to_dataframe(
+                intent_key
+            ).tp
+        elif isinstance(intent_key, pandas.core.frame.DataFrame):
+            list_keys = intent_key.tp
+        else:
+            logging.warning(
+                '''parameter list_keys must be of type
+                dialogflowcx_v3beta1.types.intent.Intent
+                or pandas.core.frame.DataFrame.
+                ''')
+            return None
+
+        if isinstance(intent_comparator, types.Intent):
+            list_comparators = Intents.intent_proto_to_dataframe(
+                intent_comparator
+            ).tp
+        elif isinstance(intent_comparator, pandas.core.frame.DataFrame):
+            list_comparators = intent_comparator.tp
+        else:
+            logging.warning(
+                '''parameter intent_comparator must be of type
+                dialogflowcx_v3beta1.types.intent.Intent
+                or pandas.core.frame.DataFrame.
+                ''')
+            return None
+
+        it = 0
+        completed = 0.0
+        tp_distances = {}
+
+        for line1 in list_keys:
+
+            phrase_similarity_list = {}
+
+            if not silent:
+                completed = float(it/len(list_keys))*100.0
+                print(
```

Comment:
@Omerside consider updating these `print` statements to `logging.info()` for consistency sake

Reply:
There's actually a reason why it's print and not logging. The logging functionality forces a newline character with every entry, whereas with print I was able to overwrite to predesignated newline at the end of a string with a blank. This allows me to update the percentage of completion without printing many lines. Will this justification suffice to keep it in place?

I did make sure that it's commented in the code, and that places where this overwrite functionality is not needed are using logging.